### PR TITLE
 Using Bundle-SymbolicName in Catalog

### PR DESCRIPTION
**Pull request by bostko**

### CODE REVIEW

File: guide/ops/catalog/index.md [hunk 1]
```diff
@@ -149,6 +149,12 @@ The following optional catalog metadata is supported:
   and other metadata (such as versions, etc) may not be applied.
 - `brooklyn.libraries`: a list of pointers to OSGi bundles required for the catalog item.
   This can be omitted if blueprints are pure YAML and everything required is included in the classpath and catalog.
+  Each pointer to an OSGi bundle can be a string representing the JAR file url or a map.
+  A map consists of:
+    - `url: http://some.server.or.other/path/my.jar` Resource where JAR file is retrieved . Usually it is with prefix `http://` or ``file://`
+    - `name: myJar-BundleSymbolicName` If not supplied here it will be obtained from <br>`Bundle-SymbolicName` written in `my.jar:META-INF/MANIFEST.MF`
+    - `version: 1.0` If not supplied here it will be obtained from <br>`Bundle-Version` written in `my.jar:META-INF/MANIFEST.MF`
```
Comment (neykov) [inline guide/ops/catalog/index.md]: Don't include this section. There's no good reason why one would need the expanded variant.

File: guide/ops/catalog/index.md [hunk 2]
```diff
@@ -161,6 +167,30 @@ The following optional catalog metadata is supported:
...
+  - id: use-from-my-catalog
+    version: "1.0.0"
+    itemType: entity
+    iconUrl: classpath://myJar-BundleSymbolicName:org/apache/brooklyn/MyEntity/icon.png
+    item:
+      type: myJar-BundleSymbolicName:org.apache.brooklyn.MyEntity
+      brooklyn.config:
+        some.config: "some value"
+        templates.install:
+          classpath://myJar-BundleSymbolicName:org/apache/brooklyn/MyEntity/config.xml: config.xml
```
Comment (neykov) [inline guide/ops/catalog/index.md]: We should try to fix cases where the prefix is needed. Does the example work without the prefixes? If not can you file a bug against Brooklyn.

File: guide/ops/catalog/index.md [hunk 3]
```diff
@@ -120,14 +120,14 @@ brooklyn.catalog:
...
+  - classpath://myJar-BundleSymbolicName:my-catalog-entries-inside-jar.bom
   - some-property: value
-    include: classpath://more-catalog-entries-inside-jar.bom
+    include: classpath://myJar-BundleSymbolicName:more-catalog-entries-inside-jar.bom
   - id: use-from-my-catalog
     version: "1.0.0"
     itemType: entity
     item:
-      type: some-type-defined-in-my-catalog-entries
+      type: myJar-BundleSymbolicName:some-type-defined-in-my-catalog-entries
```
Comment (neykov) [inline guide/ops/catalog/index.md]: Same as previous comment, prefixes shouldn't be required. Actually should be discouraged and used as a last resort to get things working.
File bugs against Brooklyn if you find cases where they are needed.

**Comment by neykov** [comment]
Would prefer to include usage of the prefix in a troubleshooting section or similar, not as the go to way of using types/resources.

File: guide/ops/catalog/index.md (hunk 2 above)
Comment (ahgittin) [inline guide/ops/catalog/index.md]: i've just put together a wee test in https://github.com/apache/brooklyn-server/pull/485 ... classpath resolution seems good for the most part

so tidy this line, and lines 181 and 185

(though we may need a fix to make 185 work, using the right `ResourceUtils` to load the `iconUrl`)

Comment (ahgittin) [inline guide/ops/catalog/index.md]: actually `iconUrl` should work fine, looks like `CatalogResourceTest` tests this

**Comment by bostko** [comment]
Bundle-Symbolic name should be discouraged. PR is irrelevant.

**Pull request closed without merging**